src/passes/shadowstack.ts

```diff
@@ -170,10 +170,27 @@ function matchPattern(module: Module, expr: ExpressionRef): ExpressionRef {
 }
 
 /** Tests whether a `value` matched by `matchTostack` needs a slot. */
-function needsSlot(module: Module, value: ExpressionRef): bool {
+function needsSlotInCall(module: Module, value: ExpressionRef): bool {
```

Comment:
Instead of duplicating the function, `needsSlot` could take a third argument `isCallOperand: bool = false` perhaps, then returning `!isCallOperand` on local gets. Wdyt?